### PR TITLE
WIP: build-test pipeline enhancement

### PR DESCRIPTION
**Pull request by abitrolly**
## Issue

#390

## Preliminary plan

* [ ] make sure Travis runs the same things as usual workflow
* [ ] ask questions to implement things in Bazel or different way.

* [ ] rewrite Shell scripts in Bash - simplify code and memorize the build logic
* [x] wrap README.md instructions in a simple script

* [ ] wrap hole process build into Docker

Local testing of multiple containers is probably better done with Kubernetes +
Ansible running on LXD. This workflow is only tested on Ubuntu right now.

### CODE REVIEW

**Comment by LeoHChen** [comment]
Can one of the admins verify this patch?

**Comment by abitrolly** [comment]
@LeoHChen it is not ready yet. Need to see Travis response.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,26 @@
 # Harmony
-[![Build Status](https://travis-ci.com/harmony-one/harmony.svg?token=DnoYvYiTAk7pqTo9XsTi&branch=master)](https://travis-ci.com/harmony-one/harmony)
+[![Build Status](https://travis-ci.org/harmony-one/harmony.svg?branch=master)](https://travis-ci.org/harmony-one/harmony)
```
Comment (harmony-ek) [inline README.md]: Nice catch (and mea culpa; I was the one who switched Harmony to travis-ci.org).  :-)

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+ROOT=$(dirname $(readlink -f "$0"))
+export GOPATH=${ROOT}/.build
+
+git submodule update --init --recursive
+
+echo "Building in .build/ subdir"
+echo "GOPATH=$GOPATH"
+
+CGO_CFLAGS="-I$GOPATH/src/github.com/harmony-one/bls/include -I$GOPATH/src/github.com/harmony-one/mcl/include -I/usr/local/opt/openssl/include"
```
Comment (harmony-ek) [inline build.sh]: We have `scripts/setup_bls_build_flags.sh` for this.  Let's concentrate the logic there (per DRY).

**Comment by harmony-ek** [comment]
BTW thanks for taking on this!  We desperately needed a smoother build pipeline :-) <3

File: README.md (hunk 1 above)
Comment (abitrolly) [reply README.md]: Maybe it is time to switch back - https://docs.travis-ci.com/user/migrate/open-source-on-travis-ci-com/ :)

**Comment by alajko** [comment]
@abitrolly Are you planning to work on this soon? Or should we close this PR.

**Pull request closed without merging**